File: API/Entities/Teacher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using API.Extensions;

namespace API.Entities
{
    public class Teacher : AppUser
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public string InsNum { get; set; }
        public string Email { get; set; }
        public string Country { get; set; }        
        public string Gender { get; set; }
        public DateTime DateOfBirth { get; set; }
        public DateTime Created { get; set; }=DateTime.Now;
        public DateTime LastActive { get; set; }=DateTime.Now;
        public ICollection<Course> Courses { get; set; }
        public ICollection<TeacherPhoto> Photos { get; set; }
        public int GetAge(){
            return DateOfBirth.CalculateAge();
        }
    }
}

File: client/src/app/_models/teacher.ts
import { Course } from "./course";

    export interface Teacher {
        id: number;
        username: string;
        name: string;
        surname: string;
        insNum: string;
        photoUrl: string;
        email: string;
        country: string;
        gender: string;
        age: number;
        created: Date;
        lastActive: Date;
        faculty: string;
        courses: Course[];
        photos: TeacherPhoto[];
    }

    export interface TeacherPhoto {
        id: number;
        url: string;
        isMain: boolean;
    }


